File: targets/js/sample-counter-v2/src/mvu/widgets/text.ts
/** biome-ignore-all lint/complexity/noUselessConstructor: explicit shape preserved by transpiler */
import type { IWidget } from "#/mvu";

export class Text implements IWidget {
  private readonly _content: string;

  constructor(content: string) {
    this._content = content;
  }

  build(): HTMLElement {
    const span = document.createElement("span");
    span.textContent = this._content;

    return span;
  }
}

File: samples/SampleCounterV2/Mvu/Widgets/Text.cs
using Metano.TypeScript.DOM;

namespace SampleCounterV2.Mvu.Widgets;

public sealed class Text(string content) : IWidget
{
    public HtmlElement Build()
    {
        var span = Js.Document.CreateElement(HtmlElementType.Span);
        span.TextContent = content;

        return span;
    }
}
